Standards/scs-0217-v1-cluster-hardening.md

```diff
@@ -0,0 +1,475 @@
+| 10259     | TCP      | kube-scheduler          | Self                  | Internal           |
+
+Hint: `Self` in the `Used by` context means, that a resource will access its own port for requests.
+
+#### Worker nodes
+
+| Ports       | Protocol | Purpose               | Used by               | Access type        |
+|-------------|----------|-----------------------|-----------------------|--------------------|
+| 10250       | TCP      | Kubelet API           | Self, Control plane   | Internal           |
+| 10255       | TCP      | Read-only Kubelet API | External applications | External, internal |
+| 30000-32767 | TCP      | NodePort Services     |                       | External           |
+
+### API security, authentication and authorization
+
+In order to secure Kubernetes against bad actors, limiting and securing access to API requests
+is recommended, since requests to those are able to control the entire Kubernetes cluster.
+Access control is applied to both human users and Kubernetes service accounts, which goes through
+several stages after a request reaches the API.
+
+1. The Kubernetes API server listens on port 6443 on the first non-localhost network interface by default,
+protected by TLS [3][controlling-access]. The TLS certificate can either be signed with a private CA or based on a public key
+infrastructure with a widely recognized CA behind it.
+2. The authentication step checks the request for correct authentication based on different possible
+authentication modules like password, plain tokens or JWT. Only one of these methods needs to succeed
+in order to allow a request to pass to the next stage.
+3. The authorization step authorizes a request, if a user is allowed to carry out a specific operation.
+The request must contain the username of the requester, the requested action and the affected object.
+Kubernetes supports different authorization modules like ABAC, RBAC or Webhooks. Only one of these
+modules need to approve the request in order for it to be authorized.
+4. The last step are Admission control modules, which can modify or reject requests after accessing
+the objects contents.
+
+#### Authentication
+
+Kubernetes provides different internal authentication mechanisms, that can be used depending
+on the requirements of the cluster provider and user. Multiple authentication systems can
+be enabled and the [Kubernetes documentation][kubernetes-auth] recommends at least using two methods,
+including Service Account Tokens and another method. Methods directly provided by Kubernetes include
+the following (a more complete or up-to-date list may be found in the [Kubernetes authentication docs][kubernetes-auth]):
+
+- *Static Token Files*
+
+  This method reads bearer tokens from requests and checks them against a CSV file provided to Kubernetes containing
+  three columns named `token`, `username` and `uid`. These tokens last indefinitely and the list cant be changed without a
```

Comment:
```suggestion
  three columns named `token`, `username` and `uid`. These tokens last indefinitely and the list can't be changed without a
```

Reply:
Updated in a2adca5b18c82097388b59a88dc665c0d2b72ab4